src/components/SearchInput/index.js

```diff
@@ -0,0 +1,137 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import classNames from "classnames";
+
+import { searchVariants } from "./constants";
+import {
+  SearchContainer,
+  StyledSearchIcon,
+  StyledInput,
+  Icon,
+  XButton,
+  Cancel,
+  Clear
+} from "./Search.styles";
+
+class SearchInput extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isFocused: !props.hasBackround
+    };
+
+    this.inputRef = React.createRef();
+    this.inputBlur = this.inputBlur.bind(this);
+    this.inputFocused = this.inputFocused.bind(this);
+  }
+
+  inputBlur() {
+    const { hasBackround, onBlur } = this.props;
+
+    if (hasBackround) {
+      this.setState({
+        isFocused: false
+      });
+    }
+
+    onBlur();
+  }
+
+  inputFocused() {
+    const { onFocus, hasBackround } = this.props;
+
+    if (hasBackround) {
+      this.setState({
+        isFocused: true
+      });
+    }
+
+    onFocus();
+  }
+
+  render() {
+    const {
+      variant,
+      placeholder,
+      value,
+      onChange,
+      clearText,
+      cancelCallback,
+      searchIconSelect,
+      className,
+      isInputVisible,
+      ...rest
+    } = this.props;
+    const { isFocused } = this.state;
+
+    return (
```

Comment:
Can you please add BEM classNames to these components? We will need selectors for automation testing

Reply:
done